dotCMS/src/main/java/com/dotcms/api/system/event/local/LocalSystemEventsAPIImpl.java

```diff
@@ -0,0 +1,205 @@
+
+    public static final String LOCAL_SYSTEM_EVENTS_THREAD_POOL_SUBMITTER_NAME = "localsystemevents.submmiter";
+
+    private final ConcurrentHashMap<Class<?>, CopyOnWriteArrayList<EventSubscriber>>  eventSubscriberByEventTypeMap;
+    private final DotSubmitter dotSubmitter;
+    private final EventSubscriberFinder eventSubscriberFinder;
+    private volatile EventSubscriber<OrphanEvent> orphanEventSubscriber;
+
+    public LocalSystemEventsAPIImpl () {
+
+        this(new ConcurrentHashMap<>(), DotConcurrentFactory.getInstance(),
+                new SubscriberAnnotationEventSubscriberFinder(),
+                new DefaultOrphanEventSubscriber());
+    }
+
+    @VisibleForTesting
+    public LocalSystemEventsAPIImpl(final ConcurrentHashMap<Class<?>, CopyOnWriteArrayList<EventSubscriber>> eventSubscriberByEventTypeMap,
+                                    final DotConcurrentFactory dotConcurrentFactory,
+                                    final EventSubscriberFinder eventSubscriberFinder,
+                                    final EventSubscriber<OrphanEvent> orphanEventSubscriber) {
+
+        this.eventSubscriberByEventTypeMap = eventSubscriberByEventTypeMap;
+        this.dotSubmitter                  = dotConcurrentFactory.getSubmitter(LOCAL_SYSTEM_EVENTS_THREAD_POOL_SUBMITTER_NAME);
+        this.eventSubscriberFinder         = eventSubscriberFinder;
+        this.orphanEventSubscriber         = orphanEventSubscriber;
+    }
+
+    @Override
+    public void subscribe(final Object subscriber) {
+
+        final  Map<Class<?>, EventSubscriber> subscriberMap =
+                this.eventSubscriberFinder.findSubscribers(subscriber);
+
+        if (null != subscriberMap && !subscriberMap.isEmpty()) {
+
+            subscriberMap.forEach((eventType, eventSubscriber) -> this.subscribe(eventType, eventSubscriber));
+        }
+    } // subscribe.
+
+    @Override
+    public void subscribe(final Class<?> eventType, final EventSubscriber subscriber) {
+
+        CopyOnWriteArrayList<EventSubscriber> eventSubscribers =
+                this.getEventSubscribersByEventType(eventType);
+
+        if (null == eventSubscribers) {
+
+            synchronized (this) {
+
+                eventSubscribers =
+                        this.getEventSubscribersByEventType(eventType);
+
+                if (null == eventSubscribers) {
+
+                    eventSubscribers = new CopyOnWriteArrayList<>();
+                    this.eventSubscriberByEventTypeMap.put(eventType, eventSubscribers);
+                }
+            }
+        }
+
+        eventSubscribers.add(subscriber);
+    } // subscribe.
+
+    @Override
+    public boolean unsubscribe(final Class<?> eventType) {
+
+        if (null != this.getEventSubscribersByEventType(eventType)) {
+
+            return null != this.eventSubscriberByEventTypeMap.remove(eventType);
+        } else {
+
+            Logger.info(this, "No any subscribers, associated to the event type: " + eventType);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "No any subscribers, associated to the event type: " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)